TypeDBConsole.java

```diff
@@ -309,6 +313,7 @@ private boolean transactionREPL(TypeDBClient client, String database, TypeDBSess
                         break;
                     } else if (replCommand.isSource()) {
                         runSource(tx, replCommand.asSource().file(), replCommand.asSource().printAnswers());
+                        hasUncommittedChanges = true;
```

Comment:
eh? what if the file only has `match` queries, then there's not uncommitted right?

Reply:
Yeah, I'd assumed it was hard to verify that condition, but on second inspection we can do it.